### DISCUSSION TITLE
Trying to understand querying vs copying with large parquet files and delta tables

### DISCUSSION THREAD

**Discussion by DrewScatterday** [duckdb/duckdb#14302] [Q&A]
## Context
- I have decently large parquet files (100+ GB) stored in azure blob storage (some tables are parquet files and some are delta tables).
- At a high level I'm writing a server (using an existing framework) that can expose these data tables as an API where users can will go into a browser and fetch data as they interact/drag around a map (using my API)
- I have been using duckDB as the database engine for this API server and it works great but I had some questions on very large datasets

## Examples: 
Query 1: 
```sql 
CREATE TABLE taxi AS 
SELECT *  
FROM read_parquet('myfile/*.parquet', hive_partitioning = true);
```

Query 2: 
```sql 
SELECT vendorID  
FROM read_parquet('myfile/*.parquet', hive_partitioning = true);
```

Query 3: 
```sql 
SELECT *  
FROM delta_scan('abfss://pathtomy/deltaTable')
WHERE vendorID = 5;
```

## Questions:
1. For query 1 since I'm creating a table and selecting all columns this would load the whole parquet table into my machines memory and then spill to disk as `.tmp` directory if it overflowed, correct?
2. With the second and third queries - duckdb queries them on the fly correct? Thus the entire table is not loaded into memory? Would this be feasible to continue querying on the fly if in a API server environment in my case or would this introduce too much latency? 

I'm trying to figure out the best way to handle this larger files that don't fit into memory and considering just using something like azure databricks SQL engine API and querying my data that way for very large datasets

**Comment by djouallah** [comment]
just my two cents, if the data does not change too often, i think it is better to import it to duckdb native file, you will get a substantial speed improvement and it has a buffer pool

**Reply by DrewScatterday** [reply]
Thanks I will look into this!
